pkg/stream/client.go

```diff
@@ -693,7 +694,8 @@ func (c *Client) BrokerLeader(stream string) (*Broker, error) {
 	streamMetadata.Leader.advHost = streamMetadata.Leader.Host
 
 	// see: https://github.com/rabbitmq/rabbitmq-stream-go-client/pull/317
-	_, err := net.LookupIP(streamMetadata.Leader.Host)
+	res := &net.Resolver{}
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
	defer cancel()
	_, err := res.LookupIPAddr(ctx, streamMetadata.Leader.Host)
```

I think it makes sense to add a 30-second timeout to the DNS lookup. This will prevent slow or unresponsive DNS servers from blocking the service indefinitely.